src/main/java/net/neoforged/neoforge/client/event/ConfigureMainRenderTargetEvent.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import com.mojang.blaze3d.pipeline.MainTarget;
+import com.mojang.blaze3d.pipeline.RenderTarget;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired when configuring the main {@linkplain RenderTarget render target}.
+ * <p>
+ * This event fires during startup when the {@link MainTarget} is constructed.
+ * <p>
+ * This event is not {@linkplain ICancellableEvent cancellable}.
+ * <p>
+ * This event is fired on the mod-speciffic event bus, only on the {@linkplain LogicalSide#CLIENT logical client}.
+ */
+public class ConfigureMainRenderTargetEvent extends Event implements IModBusEvent {
+    private boolean useDepth;
```

Comment:
The reason why I kept it is because vanilla doesn't exactly use the depth buffer themselves*, so they *may* change it to false in The Future:tm: for VRAM purposes. Furthermore, it allows mods to be explicit about whether they need the depth information to be written. (I see as this a good thing)

The defaults are basically lifted from vanilla code currently, though it may be better to inline them where the event is actually constructed

Reply:
This functionality should be removed for now.  We can re-assess if Mojang changes something or there is a formal feature request to allow it.

Reply:
Instead of removing it completely, I instead changed the patch to pass this boolean, as well as the desired width and height to the event. As we always pass `true`, I made it final and removed the set/enable method.